rs/consensus/src/consensus/metrics.rs

```diff
@@ -259,6 +262,12 @@ impl FinalizerMetrics {
                 // 1ms, 2ms, 5ms, ..., 10s, 20s, 50s
                 decimal_buckets(-3, 1),
             ),
+            finalization_latency: metrics_registry.histogram(
+                "consensus_finalization_latency_seconds",
+                "Wall time duration between block making and block finalization, in seconds",
```

Comment:
> But (not knowing much about how this round start is used), wouldn't it be noisier? I.e. is block time somehow derived from round start? And does the difference only consist of e.g. adding 1 ns to ensure strictly increasing block times

block time is not explicitly derived from the round start but a **rank-0** block is made as soon as the round starts from the rank-0 block maker's point of view; the block time is simply the block maker's wall time at the time of creating the block (unless their clock is lagging behind: https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/consensus/src/consensus/block_maker.rs?L241-251).

> or could they differ by a lot? 

for rank-0 blocks I don't think so. For higher rank block- yes, because we propose rank-i block only after `i * UNIT_DELAY` seconds have elapsed (for app subnets `UNIT_DELAY` is set to `1`, for system subnets to `3`) since the round started; so the difference can be measured in seconds.

> What I'm interested in is literally how long it took from the time the block was created until it was finalized / delivered (i.e. the time it takes the average message to get through Consensus).

sure, if that's what you're interested in then it's okay; what I proposed is different but I think both metrics make sense